### PR TITLE
Explore storing pagetab json in a Lucene index

### PR DESCRIPTION
**Pull request by behrangi**
[#186272411]

### CODE REVIEW

File: src/main/java/uk/ac/ebi/biostudies/config/IndexManager.java [hunk 1]
```diff
@@ -227,36 +237,47 @@ public void openMainIndex() throws Throwable {
         indexDirectory = FSDirectory.open(Paths.get(indexDir));
```
Comment (awaisathar) [inline src/main/java/uk/ac/ebi/biostudies/config/IndexManager.java:237]: This should probably be renamed to searchIndex

**Review by awaisathar — CHANGES_REQUESTED** [review]
Perhaps it makes sense to use the pagetab index as the primary source of pagetab for rendering and to update it when we get a message in RabbitMQ. Please see comments.

File: src/main/java/uk/ac/ebi/biostudies/service/impl/SearchServiceImpl.java [hunk 2]
```diff
@@ -258,19 +260,42 @@ public String search(String queryString, JsonNode selectedFacetsAndFields, Strin
...
+        if (fillPagetabIndex) {
+            try {
+                inputStream = getPagetabFromPagetabIndex(accession);
+            } catch (Exception ex) {
+                codeIOException = ex;
+                logger.error("problem in searching pagetab index", ex);
+            }
+        }
+        try {
+            if (inputStream == null) {
```
Comment (awaisathar) [inline src/main/java/uk/ac/ebi/biostudies/service/impl/SearchServiceImpl.java:284]: We should push the document in the index when we load it from NFS/FIRE

**Review by awaisathar — APPROVED** [review]
Let's cache the document in pagetab index when we load it from FIRE/NFS

File: src/main/java/uk/ac/ebi/biostudies/service/impl/SearchServiceImpl.java (hunk 2 above)
Comment (behrangi) [reply src/main/java/uk/ac/ebi/biostudies/service/impl/SearchServiceImpl.java:284]: So do you agree to remove saving pagetab in indexone service and just keep this cache gradually filling service?
I modified it some how that add pagetab from mongodb dump and indexone message in the same mechanism. Is this fine? (as you said the mongodb pagetab has some missing values)

Comment (awaisathar) [reply src/main/java/uk/ac/ebi/biostudies/service/impl/SearchServiceImpl.java:284]: MongoDB documents are dumps of the internal model and do not follow the public PageTab specs. This new index should be for public PageTab only (available on FTP/Aspera/Globus etc) so we should probably add the public version to the index after loading it from NFS/FIRE.

**Pull request merged**